test/scalarstats.jl

```diff
@@ -17,6 +17,13 @@ using Base.Test
 @test_approx_eq harmmean(1:3) 3 / (1 + 1/2 + 1/3)
 @test_approx_eq harmmean([1, 2, 4]) 12 / 7
 
+## genmean
+@test_approx_eq genmean([1,1,2,3], 1) 7/4
+@test_approx_eq genmean([1,4,2], -1) 12/7
+@test_approx_eq genmean([1,1,2,3], 0) (6.0)^(1/4)
+@test_approx_eq_eps genmean([1,1,2,3], 1e-8) (6.0)^(1/4) 1e-8
```

Comment:
Trying to ensure that the algorithm is numerically stable when `p` is near to 0 (answer should be very close to the geometric mean).